crates/node/src/migration_service/web.rs

```diff
@@ -0,0 +1,270 @@
+            let migration_state_receiver_clone = migration_state_receiver.clone();
+            let tls_acceptor = tls_acceptor.clone();
+            let state_clone = state.clone();
+            tokio::spawn(async move {
+                tracing::info!("Handle connection");
+                let stream = tls_acceptor.accept(tcp_stream).await?;
+                let client_public_key = mpc_tls::tls::extract_public_key(stream.get_ref().1)?;
+                authenticate_backup_service(&migration_state_receiver_clone, client_public_key)?;
+                tracing::info!("TLS handshake complete, backup service authenticated and encrypted channel established.");
+                tokio::spawn(async move {
+                    if let Err(err) = hyper::server::conn::Http::new()
+                        .serve_connection(
+                            stream,
+                            service_fn(move |req| handle_with_state(req, state_clone.clone())),
+                        )
+                        .await
+                    {
+                        tracing::error!("Error serving connection: {err}");
+                    }
+                });
+                anyhow::Ok(())
+            });
+        }
+    });
+
+    tracing::info!(address = %bind_address,"Successfully bound to address");
+    Ok(())
+}
+
+#[allow(dead_code)]
+/// Connects to the web server, performs the TLS handshake and sends a simple HTTP request.
+pub async fn connect_to_web_server(
+    p2p_private_key: &ed25519_dalek::SigningKey,
+    target_address: &str,
+    expected_server_key: VerifyingKey,
+) -> anyhow::Result<String> {
+    let (_server_config, client_config) = configure_tls(p2p_private_key)?;
+    let conn = TcpStream::connect(target_address)
+        .await
+        .context("TCP connect")?;
+    let tls_conn = tokio_rustls::TlsConnector::from(Arc::new(client_config))
+        .connect("dummy".try_into().unwrap(), conn)
+        .await
+        .context("TLS connect")?;
+
+    let public_key = mpc_tls::tls::extract_public_key(tls_conn.get_ref().1)?;
+    if public_key != expected_server_key {
+        tracing::info!(
+            ?expected_server_key,
+            ?public_key,
+            "closing connection, public key mismatch"
+        );
+        anyhow::bail!("closing connection, public key mismatch");
+    }
+
+    tracing::info!(
+        "TLS handshake complete, backup service authenticated and encrypted channel established."
+    );
+
+    let (mut request_sender, connection) = hyper::client::conn::handshake(tls_conn)
+        .await
+        .context("failed to perform HTTP handshake")?;
+
+    // Run the connection driver in the background
+    tokio::spawn(async move {
+        if let Err(err) = connection.await {
+            tracing::error!("Connection error: {err}");
+        }
+    });
+
+    let req = Request::builder()
+        .method("GET")
+        .uri("https://example/hello") // doesn’t matter, server matches on path only
```

Comment:
can could also use here something like `http://{target_address}/hello`  (makes explicit this is an HTTP connection), or is the correct way to use `https` ? I understand that the server address does not matter as we are already inside the TLS tunnel.